extensions/ql-vscode/src/databases/config/db-config-store.ts

```diff
@@ -76,6 +78,84 @@ export class DbConfigStore extends DisposableObject {
     await this.writeConfig(config);
   }
 
+  public async removeDbItem(dbItem: DbItem): Promise<void> {
+    if (!this.config) {
+      throw Error("Cannot remove item if config is not loaded");
+    }
+
+    const config: DbConfig = cloneDbConfig(this.config);
+    const selectedItem: SelectedDbItem | undefined = config.selected;
+
+    // Remove item from databases
+    switch (dbItem.kind) {
+      case DbItemKind.LocalList:
+        config.databases.local.lists = config.databases.local.lists.filter(
+          (list) => list.name !== dbItem.listName,
+        );
+        break;
+      case DbItemKind.RemoteUserDefinedList:
+        config.databases.remote.repositoryLists =
+          config.databases.remote.repositoryLists.filter(
+            (list) => list.name !== dbItem.listName,
+          );
+        break;
+      case DbItemKind.LocalDatabase:
+        if (dbItem.parentListName) {
+          const parent = config.databases.local.lists.find(
+            (list) => list.name === dbItem.parentListName,
+          );
+          if (!parent) {
+            throw Error(`Cannot find parent list '${dbItem.parentListName}'`);
+          } else {
+            parent.databases = parent.databases.filter(
+              (db) => db.name !== dbItem.databaseName,
+            );
+          }
+        }
+        config.databases.local.databases =
+          config.databases.local.databases.filter(
+            (db) => db.name !== dbItem.databaseName,
+          );
+        break;
+      case DbItemKind.RemoteRepo:
+        if (dbItem.parentListName) {
+          const parent = config.databases.remote.repositoryLists.find(
+            (list) => list.name === dbItem.parentListName,
+          );
+          if (!parent) {
+            throw Error(`Cannot find parent list '${dbItem.parentListName}'`);
+          } else {
+            parent.repositories = parent.repositories.filter(
+              (repo) => repo !== dbItem.repoFullName,
+            );
+          }
+        }
+        config.databases.remote.repositories =
+          config.databases.remote.repositories.filter(
+            (repo) => repo !== dbItem.repoFullName,
+          );
+        break;
+      case DbItemKind.RemoteOwner:
+        config.databases.remote.owners = config.databases.remote.owners.filter(
+          (owner) => owner !== dbItem.ownerName,
+        );
+        break;
+      default:
+        throw Error(`Type '${dbItem.kind}' cannot be removed`);
+    }
+
+    // Remove item from selected
+    const mappedItem = mapDbItemToSelectedDbItem(dbItem);
+    if (
+      selectedItem &&
+      JSON.stringify(mappedItem) === JSON.stringify(selectedItem)
```

Comment:
Open to suggestions!! This is a very ugly solution that I would fortify with writing tests for all cases - I experimented with other options but got errors and didn't want write 100 lines of code just to compare these objects. 

Also: if this check fails and the selected item is not removed it will go unnoticed in most cases and if the edge case happens that a user runs a query directly after removing the selected item they will get an error which should be a minor inconvenience only.

Reply:
Are we trying to be too generic? If the code was in the various switch cases, the shape of the `SelectedDbItem` would be known and easy to compare I think.

I'm having to do similar checks in the area I'm working on so we could go with something for now and circle back to it too if it helps.

Reply:
I found good solution using a library we already have.